tests/integration/140.command.sites.test.js

```diff
@@ -0,0 +1,61 @@
+const test = require('ava')
+const execa = require('execa')
+const stripAnsi = require('strip-ansi')
+
+const cliPath = require('./utils/cli-path')
+const { CONFIRM, answerWithValue, handleQuestions } = require('./utils/handle-questions')
+const { withMockApi } = require('./utils/mock-api')
+// TODO: Flaky tests enable once fixed
+/**
+ * As some of the tests are flaky on windows machines I will skip them for now
+ * @type {import('ava').TestInterface}
+ */
+
+test.skip('netlify sites:create-template', async (t) => {
+  const siteTemplateQuestions = [
```

Comment:
@lukasholzer I tried to add some tests for this command but I've been struggling and could do with some guidance on how to mock the API calls to Github if you have some idea or if you know somewhere else in the code I should look at, please? 🙏  I tried using `sinon` and also [`nock`](https://github.com/nock/nock) but didn't manage to make it work 😞 

For now I skipped these tests cause they ping the real Github API.

Reply:
Hey @charliegerard I've added a commit showcasing how to mock the github api – so you should good to go writing the test.

The problem you faced was that it is impossible to mock something that is spawned in a child process -> what you do by calling `execa` -> basically a wrapper over `child_process` `spawn`.

So the cool thing is that you can start up a command without invoking it as a child process and then you achieve ultimate power of having the same global context and therefore mocking things out.

The only importance is that you need to do the mocks before you require the actual code. As it needs to patch the functions first with the stubbed logic.

I hope you are good to go now :) otherwise don't hesitate to reach out to me maybe we can do a 🍐  on that if it helps you!

Reply:
Here the commit: https://github.com/netlify/cli/pull/3948/commits/aee0ff51c4230aa5601d870db96fdb7b6eda5e0e

Reply:
Awesome, thank you so much! I'll have a look!